secrethub/resource_secret.go

```diff
@@ -108,7 +108,7 @@ func resourceSecretCreate(d *schema.ResourceData, m interface{}) error {
 			charset = randchar.Alphanumeric
 		}
 		if useSymbols {
-			charset = randchar.All
+			charset = charset.Add(randchar.Symbols)
```

Comment:
Saw this come past and read Simon's comment. We're deprecating the `use_symbols` field, right?

Then the reason to keep `use_symbols` functionality in place wouldn't be to have it work with together the `charset` functionality, but to avoid breaking existing code. 

Now consider your change. Does it alter the behavior of existing terraform code? As in, does setting `use_symbols` with this new version use the exact same underlying charset as before?

That, IMO, should be the point of this code. If both work together, that's fine, but there's also something to be said for not allowing deprecated and new fields to be used together, as allowing it creates more unpredictable situations. 

Not saying we should choose A or B, but I'd like us to be conscious as to why we're choosing one or the other and what the implications are of that choice.

Reply:
Yes, we could also choose to return an error when both options are supplied. I think supporting to use them together is simpler in this case.